drivers/pinctrl/pfc_rcar.c

```diff
@@ -21,20 +21,24 @@
 #if defined(CONFIG_SOC_SERIES_RCAR_GEN3)
 #define PFC_RCAR_GPSR 0x100
 #define PFC_RCAR_IPSR 0x200
-DEVICE_MMIO_TOPLEVEL_STATIC(pfc, DT_DRV_INST(0));
-static uintptr_t reg_base[1];
 #elif defined(CONFIG_SOC_SERIES_RCAR_GEN4)
 #define PFC_RCAR_GPSR 0x040
 #define PFC_RCAR_IPSR 0x060
-/* swap both arguments */
-#define PFC_REG_ADDRESS(idx, node_id) DT_REG_ADDR_BY_IDX(node_id, idx)
-static const uintptr_t reg_base[] = {
-	LISTIFY(DT_NUM_REGS(DT_DRV_INST(0)), PFC_REG_ADDRESS, (,), DT_DRV_INST(0))
-};
 #else
 #error Unsupported SoC Series
 #endif
 
+/* swap both arguments */
+#define PFC_REG_ADDRESS(idx, inst) DT_INST_REG_ADDR_BY_IDX(inst, idx)
+static uintptr_t reg_base[] = {
```

Comment:
const here?

Reply:
nope, the value can be changed by device_map call during the driver init